plugins/chat/admin/assets/javascripts/admin/components/chat-incoming-webhook-edit-form.gjs

```diff
@@ -0,0 +1,192 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { fn } from "@ember/helper";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import { not } from "truth-helpers";
+import DButton from "discourse/components/d-button";
+import EmojiPicker from "discourse/components/emoji-picker";
+import Form from "discourse/components/form";
+import replaceEmoji from "discourse/helpers/replace-emoji";
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import i18n from "discourse-common/helpers/i18n";
+import I18n from "discourse-i18n";
+import ChatChannelChooser from "discourse/plugins/chat/discourse/components/chat-channel-chooser";
+
+export default class ChatIncomingWebhookEditForm extends Component {
+  @service toasts;
+  @service router;
+  @tracked emojiPickerIsActive = false;
+  @tracked webhook;
+
+  constructor() {
+    super(...arguments);
+    this.webhook = this.args.webhook;
```

Comment:
do you really need this? if you do you, can just do outside of the constructor:

`webhook = this.args.webhook;`

I generally avoid doing this, even if slightly more verbose because it creates a confusion between what is external to the component and what is internal. You could easily do `this.webhook = "xxx";` which would be a big mistake and a slightly hard to see bug. Also it means you can remove the `@tracked webhook;`

Reply:
Yes I don't need it now, will remove. This was trying to avoid route reloads and such